src/components/preference-item/preference-item.component.spec.ts

```diff
@@ -1,37 +1,112 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { PreferenceItemComponent } from './preference-item.component';
 import { BehaviorSubject } from 'rxjs';
+import { Preference, PreferenceWithMeta } from '../../services/preference/types';
+import { MatCardModule } from '@angular/material/card';
+
+interface TestPreference extends Preference {
+  num: number;
+  bool: boolean;
+  select: string;
+}
 
 describe('PreferenceItemComponent', () => {
-  let fixture: ComponentFixture<PreferenceItemComponent<any>>;
-  let component: PreferenceItemComponent<any>;
+  let boolFixture: ComponentFixture<PreferenceItemComponent<TestPreference, 'bool'>>;
+  let numFixture: ComponentFixture<PreferenceItemComponent<TestPreference, 'num'>>;
+  let selectFixture: ComponentFixture<PreferenceItemComponent<TestPreference, 'select'>>;
+  let boolComponent: PreferenceItemComponent<TestPreference, 'bool'>;
+  let numComponent: PreferenceItemComponent<TestPreference, 'num'>;
+  let selectComponent: PreferenceItemComponent<TestPreference, 'select'>;
+  let preference$: BehaviorSubject<PreferenceWithMeta<TestPreference>>;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
+      imports: [
+        MatCardModule,
+      ],
       declarations: [
         PreferenceItemComponent,
       ],
     });
-    fixture = TestBed.createComponent(PreferenceItemComponent);
-    component = fixture.componentInstance;
+    boolFixture = TestBed.createComponent<PreferenceItemComponent<TestPreference, 'bool'>>(PreferenceItemComponent);
```

Comment:
Same comment here about simplifying these tests into distinct describe blocks (which can be done in a later PR if preferred).